src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * @author leslieharland
+ */
+
+package duke.task;
+
+/**
+ * The Class Deadline.
+ */
+public class Deadline extends Task {
+
+    /**
+     * Instantiates a new deadline.
+     *
+     * @param description the description
+     * @param date        the date
+     */
+    public Deadline(String description, String date) {
+        super(description, date);
+        type = "D";
+    }
+
+    /**
+     * Parses the.
+     *
+     * @param taskString the task string
+     * @return the deadline
+     */
+    public static Deadline parse(String taskString) {
+        assert (!taskString.isEmpty());
+        String[] parts = taskString.split("\\|");
+        String desc = parts[2];
+        Deadline t = new Deadline(desc, "");
+        if (parts.length > 3) {
+            t = new Deadline(desc, parts[3]);
+        }
+        t.isDone = parts[1].trim().equals("1");
+
+        return t;
+    }
+
+    /**
+     * To string.
```

Comment:
It would be better if the java comment could explain how the String returned by the toString() method is formatted. For example, "1. Deadline (by: 20 May 2020)"